src/utilcode/longfilepathwrappers.cpp

```diff
@@ -0,0 +1,1317 @@
+                ret = GetLongPathNameW(
+                        shortPath.GetUnicode(),
+                        lpszLongPath.OpenUnicodeBuffer(ret - 1),
+                        ret
+                        );
+
+            }
+
+            lpszLongPath.CloseBuffer(ret);
+        }
+        
+        lastError = GetLastError();
+    }
+    EX_CATCH_HRESULT(hr);
+
+    if (hr != S_OK )
+    {
+        SetLastError(hr);
+    }
+    else if(ret == FALSE)
+    {
+        SetLastError(lastError);
+    }
+
+    return ret;
+}
+
+BOOL
+CreateDirectoryWrapper(
+        _In_ LPCWSTR lpPathName,
+        _In_opt_ LPSECURITY_ATTRIBUTES lpSecurityAttributes
+        )
+{
+    HRESULT hr = S_OK;
+    BOOL ret   = FALSE;
+    DWORD lastError;
+
+    EX_TRY
+    {
+        LongPathString path(LongPathString::Literal, lpPathName);
+
+        if (SUCCEEDED(LongFile::NormalizePath(path)))
+        {
+            ret = CreateDirectoryW(
+                    path.GetUnicode(),
+                    lpSecurityAttributes
+                    );
+        }
+            
+        lastError = GetLastError();
+    }
+    EX_CATCH_HRESULT(hr);
+
+    if (hr != S_OK )
+    {
+        SetLastError(hr);
+    }
+    else if(ret == FALSE)
+    {
+        SetLastError(lastError);
+    }
+
+    return ret;
+
+}
+
+BOOL
+RemoveDirectoryWrapper(
+        _In_ LPCWSTR lpPathName
+        )
+{
+    HRESULT hr = S_OK;
+    BOOL ret   = FALSE;
+    DWORD lastError;
+    EX_TRY
+    {
+        LongPathString path(LongPathString::Literal, lpPathName);
+
+        if (SUCCEEDED(LongFile::NormalizePath(path)))
+        {
+            ret = RemoveDirectoryW(
+                    path.GetUnicode()
+                    );
+        }
+        
+        lastError = GetLastError();
+    }
+    EX_CATCH_HRESULT(hr);
+
+    if (hr != S_OK )
+    {
+        SetLastError(hr);
+    }
+    else if(ret == FALSE)
+    {
+        SetLastError(lastError);
+    }
+
+    return ret;
+
+}
+DWORD
+GetModuleFileNameWrapper(
+    _In_opt_ HMODULE hModule,
+    SString& buffer
+    )
+{
+    HRESULT hr = S_OK;
+    DWORD ret = FALSE;
+    DWORD lastError;
+    EX_TRY
+    {
+        COUNT_T size = buffer.GetUnicodeAllocation();
+        size = size > 20 ? size : MAX_PATH;
+
+        ret = GetModuleFileNameW(
+            hModule, 
+            buffer.OpenUnicodeBuffer(size - 1),
+            (DWORD)size
+            );
+
+        
+        while (ret == size )
+        {
+            buffer.CloseBuffer();
+            if (ret == FALSE)
+            {
+                break;
+            }
+            size = size * 2;
+            ret = GetModuleFileNameW(
+                hModule,
+                buffer.OpenUnicodeBuffer(size - 1),
+                (DWORD)size
+                );
+          
+        }
+        
+
+        lastError = GetLastError();
+        buffer.CloseBuffer(ret);
+    }
+    EX_CATCH_HRESULT(hr);
+
+    if (hr != S_OK)
+    {
+        SetLastError(hr);
+    }
+    else if (ret == FALSE)
```

Comment:
I am changing it in all the places. I was doing a mass edit and left it as is a BOOL is actually typed as an Int correct ?